mmdet/models/dense_heads/ascend_anchor_head.py

```diff
@@ -0,0 +1,385 @@
+    """  # noqa: W605
+
+    def __init__(self,
+                 num_classes,
+                 in_channels,
+                 feat_channels=256,
+                 anchor_generator=dict(
+                     type='AnchorGenerator',
+                     scales=[8, 16, 32],
+                     ratios=[0.5, 1.0, 2.0],
+                     strides=[4, 8, 16, 32, 64]),
+                 bbox_coder=dict(
+                     type='DeltaXYWHBBoxCoder',
+                     clip_border=True,
+                     target_means=(.0, .0, .0, .0),
+                     target_stds=(1.0, 1.0, 1.0, 1.0)),
+                 reg_decoded_bbox=False,
+                 loss_cls=dict(
+                     type='CrossEntropyLoss',
+                     use_sigmoid=True,
+                     loss_weight=1.0),
+                 loss_bbox=dict(
+                     type='SmoothL1Loss', beta=1.0 / 9.0, loss_weight=1.0),
+                 train_cfg=None,
+                 test_cfg=None,
+                 init_cfg=dict(type='Normal', layer='Conv2d', std=0.01)):
+        super(AscendAnchorHead, self).__init__(
+            num_classes=num_classes,
+            in_channels=in_channels,
+            feat_channels=feat_channels,
+            anchor_generator=anchor_generator,
+            bbox_coder=bbox_coder,
+            reg_decoded_bbox=reg_decoded_bbox,
+            loss_cls=loss_cls,
+            loss_bbox=loss_bbox,
+            train_cfg=train_cfg,
+            test_cfg=test_cfg,
+            init_cfg=init_cfg)
+
+    def get_batch_gt_bboxes(self, gt_bboxes_list, num_images, gt_nums, device,
+                            max_gt_labels):
+        """Get ground truth bboxes of all image.
+
+        Args:
+            gt_bboxes_list (list[Tensor]): Ground truth bboxes of each image.
+            num_images (int): The num of images.
+            gt_nums(list[int]): The ground truth bboxes num of each image.
+            device (torch.device | str): Device for returned tensors
+            max_gt_labels(int): The max ground truth bboxes num of all image.
+        Returns:
+            batch_gt_bboxes: (Tensor): Ground truth bboxes of all image.
+        """
+        if not hasattr(self, 'batch_gt_bboxes'):
+            self.batch_gt_bboxes = {}
+        if not hasattr(self, 'min_anchor'):
+            self.min_anchor = (-1354, -1344)
+        if gt_bboxes_list is None:
+            batch_gt_bboxes = None
+        else:
+            if self.batch_gt_bboxes.get(max_gt_labels) is None:
+                batch_gt_bboxes = torch.zeros((num_images, max_gt_labels, 4),
+                                              dtype=gt_bboxes_list[0].dtype,
+                                              device=device)
+                batch_gt_bboxes[:, :, :2] = self.min_anchor[0]
+                batch_gt_bboxes[:, :, 2:] = self.min_anchor[1]
+                self.batch_gt_bboxes[max_gt_labels] = batch_gt_bboxes.clone()
```

Comment:
What is the purpose of caching? This field will keep increasing during training?